dbms/src/Storages/MergeTree/MergeTreeIndexConditionBloomFilter.cpp

```diff
@@ -297,9 +300,28 @@ bool MergeTreeIndexConditionBloomFilter::traverseASTEquals(
     {
         size_t position = header.getPositionByName(key_ast->getColumnName());
         const DataTypePtr & index_type = header.getByPosition(position).type;
-        Field converted_field = convertFieldToType(value_field, *index_type, &*value_type);
-        out.predicate.emplace_back(std::make_pair(position, BloomFilterHash::hashWithField(&*index_type, converted_field)));
-        out.function = function_name == "equals" ? RPNElement::FUNCTION_EQUALS : RPNElement::FUNCTION_NOT_EQUALS;
+        const DataTypeArray * array_type = typeid_cast<const DataTypeArray *>(index_type.get());
+
+        if (function_name == "has")
+        {
+            out.function = RPNElement::FUNCTION_HAS;
+
+            if (!array_type)
+                throw Exception("First argument for function has must be an array.", ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+            Field converted_field = convertFieldToType(value_field, *array_type->getNestedType(), &*value_type);
+            out.predicate.emplace_back(std::make_pair(position, BloomFilterHash::hashWithField(&*index_type, converted_field)));
```

Comment:
Maybe `BloomFilterHash::hashWithField(&*array_type->getNestedType(), converted_field)` would be better?